includes/wizards/audience/class-audience-campaigns.php

```diff
@@ -63,4 +114,929 @@ public function add_page() {
+						'sanitize_callback' => 'absint',
+					],
+					'name' => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/wizard/' . $this->slug . '/rename-campaign/(?P<id>\w+)',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_campaign_rename' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					'id'   => [
+						'sanitize_callback' => 'absint',
+					],
+					'name' => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/wizard/' . $this->slug . '/subscription-products',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'api_get_subscription_products' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					's' => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * API endpoint callbacks
+	 */
+
+	/**
+	 * Get data to render Wizard.
+	 *
+	 * @param array|null $extras optional array of parameters to include in response.
+	 * @return WP_REST_Response
+	 */
+	public function api_get_settings( $extras = [] ) {
+		$newspack_popups_configuration_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'newspack-popups' );
+
+		$response = array_merge(
+			[
+				'prompts'   => [],
+				'segments'  => [],
+				'campaigns' => [],
+				'settings'  => [],
+			],
+			(array) $extras
+		);
+
+		if ( $newspack_popups_configuration_manager->is_configured() ) {
+			$response['prompts']   = array_map(
+				function( $prompt ) {
+					$prompt['edit_link'] = get_edit_post_link( $prompt['id'] );
+					return $prompt;
+				},
+				$newspack_popups_configuration_manager->get_prompts( true, true )
+			);
+			$response['segments']  = $newspack_popups_configuration_manager->get_segments( true );
+			$response['settings']  = $newspack_popups_configuration_manager->get_settings();
+			$response['campaigns'] = $newspack_popups_configuration_manager->get_campaigns();
+		}
+
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Set terms for one Popup.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response with the info.
+	 */
+	public function api_set_popup_terms( $request ) {
+		$id       = $request['id'];
+		$terms    = $request['terms'];
+		$taxonomy = $request['taxonomy'];
+
+		$newspack_popups_configuration_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'newspack-popups' );
+
+		$response = $newspack_popups_configuration_manager->set_popup_terms( $id, $terms, $taxonomy );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		return $this->api_get_settings();
+	}
+
+	/**
+	 * Update settings for a Pop-up.
```

Comment:
```suggestion
	 * Update settings for a prompt.
```

ABSOLUTELY UNNECESSARY NIT: is now a good time to do the ol' Find/Replace to rename `pop-up` to `prompt`? 😄

Reply:
Can't think of a better time! Done in 3a06b084b3b99934c97231d85e7288e00c434711.

I've left the CM methods unchanged since they are wrappers for the plugin's methods. The plugin still uses `popup`.

I also removed the `api_set_popup_terms()` which was unused (also unused in `trunk`).